src/main/java/core/basesyntax/db/Storage.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax.db;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
+    private static final Map<String, Integer> fruitStorage = new HashMap<>();
+
+    public static void setBalance(String fruit, int quantity) {
+        if (quantity < 0) {
+            throw new RuntimeException("Quantity can't be negative");
+        }
+        if (fruitStorage.values().stream().anyMatch(value -> value < 0)) {
+            throw new RuntimeException("Balance can't be negative");
+        }
+        fruitStorage.put(fruit, quantity);
+    }
+
+    public static void addFruit(String fruit, int quantity) {
+        if (quantity < 0) {
+            throw new RuntimeException("Quantity can't be negative");
+        }
+        if (fruitStorage.values().stream().anyMatch(value -> value < 0)) {
+            throw new RuntimeException("Balance can't be negative");
+        }
+        fruitStorage.put(fruit, fruitStorage.getOrDefault(fruit, 0) + quantity);
+    }
```

Comment:
The check for negative balance is redundant in the method for adding fruits. It's good that you are checking the quantity to ensure it's not negative, but the balance check should be in the 'purchase' operation handler.